flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -310,7 +311,10 @@ public void createCatalog(String catalogName, CatalogDescriptor catalogDescripto
         }
 
         Catalog catalog = initCatalog(catalogName, catalogDescriptor);
-        catalog.open();
+        try (TemporaryClassLoaderContext context =
```

Comment:
@jrthe42  Thanks for the contribution, Can you add a test verity this  change ?

Reply:
@hackergin thanks, a test case is added to verify the change.